main.go

```diff
@@ -18,6 +21,11 @@ var (
 )
 
 func main() {
+	if memStat, err := mem.Get(); err == nil {
+		memoryLimit := int64(80 * memStat.Total / 100)
+		logger.NewLogger().Infow("Setting memory limit to", "limit", memoryLimit)
+		debug.SetMemoryLimit(memoryLimit)
+	}
```

Comment:
**[note]** setting a soft memory limit (aka `GOMEMLIMIT`) to 80% of the container's available memory.